common/components/GasSlider/GasSlider.tsx

```diff
@@ -49,18 +51,26 @@ class GasSlider extends React.Component<Props, State> {
   }
 
   public render() {
-    const { offline, disableAdvanced, gasPrice } = this.props;
+    const { offline, disableAdvanced, gasPrice, readOnly } = this.props;
     const showAdvanced = (this.state.showAdvanced || offline) && !disableAdvanced;
 
     return (
       <div className="GasSlider">
-        {showAdvanced ? (
+        {readOnly ? (
+          <GasLimitField
```

Comment:
GasSlider should be re-named something else, and have `slider` as a boolean prop instead as having it display an input field sometimes based on the `readOnly` prop breaks semantics